src/button-group/interfaces.ts

```diff
@@ -90,7 +92,20 @@ export interface ButtonGroupProps extends BaseComponentProps {
    * * `text` (string) - The name of the group rendered as ARIA label for this group.
    * * `items` ((ButtonGroupProps.IconButton | ButtonGroupProps.MenuDropdown)[]) - The array of items that belong to this group.
    */
-  items: ReadonlyArray<ButtonGroupProps.ItemOrGroup>;
+  items?: ReadonlyArray<ButtonGroupProps.ItemOrGroup>;
+  /**
+   * Instead of passing `items`, group items can be passed as children. When using this,
+   * you will need to manage click handlers and state of buttons.
+   *
+   * Each item should contain a single focusable element which must have a unique `data-itemid` attribute.
+   *
+   * For example:
+   * - `<Button data-itemid="button-one">One</Button>`
+   * - `<ButtonDropdown nativeButtonAttributes={{'data-itemid': 'button-two'}} items={...} />`
+   *
+   * @awsuiSystem core
+   */
+  children?: ReactNode;
```

Comment:
I think this defies the purpose of the button group component, the main complexity of which lies in the way we deal with focusing, keyboard nav, and tooltips - neither of these features can be easily replicated from the outside as we don't share the related components/tools.

Should we instead think of some advanced custom item type, e.g. `{ type: 'custom', button: (props) => <MyButton {...props} />, tooltip: (props) => <MyTooltipContent {...props} /> }`?

Reply:
It still deals with focus & keyboard navigation, only tooltips are not included. But the tooltips are only really relevant for icon-only buttons, so I don't think there's so much lost by not supporting those for custom buttons which by definition are likely to be different from what's supported via `items`

Reply:
Is it possible to combine the children and items API? Otherwise, the icon-buttons added as children won't have tooltips, which is a downgrade to UX and potentially an a11y issue.

Reply:
Should we use `before` and `after` instead of children, maybe? So then one can use `items` but also append or prepend some custom actions. Alternatively, we can introduce a new item type "custom" - so one can pass it like `items: [{ id: 'custom-button', type: 'custom', content: <Button /> }]`.